concepts/eventing/performance evaluation/nats-jetstream/results_prod_HA.md

```diff
@@ -0,0 +1,332 @@
+
+---
+
+## Test Scenario 4: K8s Node deleted during test where stream leader NATS Pod was deployed (with Stream Replicas: 3)
+
+The number of stream replicas was changed to 3 replicas in order to check the behaviour of single-node failure with multiple stream replicas.
+
+```
+nats stream update sap --replicas 3 -f
+```
+
+### Run ID: 9/6/2022T8:31 (Duration: 5m, Event Rate: 150rps)
+
+In this test run, the k8s Node was deleted where `eventing-nats-1` Pod (Stream Leader) was deployed after 2 minutes of test execution. The eventing-controller pod was also deployed on the deleted node and it was re-scheduled after the crashed Node came back online.
+
+**State before test run:**
+- Stream:
+  - Leader: eventing-nats-1
+  - Replicas: 3
+
+**State after test run:**
+- Stream:
+  - Leader: eventing-nats-2
+  - Replicas: 3
+
+- Stream info after deleting the K8s Node:
+  ```
+  ╰─ nats stream info sap
+  Information for Stream sap created 2022-06-09T10:01:02+02:00
+  
+  Configuration:
+  
+               Subjects: kyma.>
+       Acknowledgements: true
+              Retention: File - Limits
+               Replicas: 3
+         Discard Policy: Old
+       Duplicate Window: 2m0s
+      Allows Msg Delete: true
+           Allows Purge: true
+         Allows Rollups: false
+       Maximum Messages: unlimited
+          Maximum Bytes: unlimited
+            Maximum Age: unlimited
+   Maximum Message Size: unlimited
+      Maximum Consumers: unlimited
+  
+  
+  Cluster Information:
+  
+                   Name: eventing-nats
+                 Leader: eventing-nats-2
+                Replica: eventing-nats-1, outdated, seen 29.48s ago, 18,466 operations behind
+                Replica: eventing-nats-0, current, seen 0.00s ago
+  
+  State:
+  
+               Messages: 18,463
+                  Bytes: 12 MiB
+               FirstSeq: 1 @ 2022-06-09T08:31:02 UTC
+                LastSeq: 18,463 @ 2022-06-09T08:33:15 UTC
+       Active Consumers: 1
+  ```
+
+**Test Results:**
+```
+-> Total Events Sent      by **Test Sender** : 43,083 (+ 1 Failed = 43,084)
+-> Total Events Received  by **Sink**        : 43,012 (There were some errors in sink for storing the data to database)
+```
+
+**Test Results Dashboard:**
+
+![](assets/ha_crash4_1.png "")
+
+**Test Sender Stats:**
+
+![](assets/ha_crash4_2.png "")
+
+**Finding:** 
+
+As now there were 3 stream replicas, deleting the K8s Node where stream leader NATS Pod was deployed triggered a leader election for the stream. During the election time, event-publisher-proxy failed to publish some events to NATS. But this downtime is much less than the `Test Scenario 2` where there was only one stream replica and it had to wait for the same NATS Pod to come up in order to resume receiving events. Instead, in this scenario a new NATS Pod was elected as the leader for the stream.
```

Comment:
```suggestion
Because now there were three stream replicas, deleting the Kubernetes Node in which the stream leader NATS Pod was deployed triggered a leader election for the stream. During the election time, event-publisher-proxy failed to publish some events to NATS. 
But this downtime is much shorter than in test scenario 2 where there was only one stream replica and it had to wait for the same NATS Pod to come up in order to resume receiving events. Instead, in test scenario 4, a new NATS Pod was elected as the leader for the stream.
```